### PR TITLE
[#11] Sequence Picker

### LINKED ISSUE #11: Position sequence picker
An async application that allows a user to pick a limited amount of positions on the canvas

### PR DESCRIPTION
**Pull request by krbz999**
Closes #11

Also fixes some syntax.

Example usage:
```js
const ret = await polo.utils.applications.sequence.create({links: 5});
console.warn(ret);
for(const point of ret) {
  console.warn(point);
  await canvas.ping(point);
}
```

### CODE REVIEW

File: scripts/applications/sequence-picker.mjs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    return game.i18n.localize("POLO.SequencePicker");
+  }
+
+  /** @override */
+  static get defaultOptions() {
+    return foundry.utils.mergeObject(super.defaultOptions, {
+      resizeable: false,
+      minimizable: false,
+      template: "modules/polo-utils/templates/applications/sequence-picker.hbs",
+      classes: ["sequence-picker"],
```
Comment (Freeze023) [inline scripts/applications/sequence-picker.mjs]:
```suggestion
      classes: ["polo-utils-sequence-picker"],
```

File: scripts/applications/sequence-picker.mjs [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+  }
+
+  /** @override */
+  async render(...args) {
+    this._destroySprites();
+    const points = this.model.sequence;
+    const img = "icons/sundries/flags/banner-flag-blue.webp";
+    this.sprites = new Set();
+    for (const point of points) {
+      const spr = PIXI.Sprite.from(img, {scale: 0.5});
```
Comment (Freeze023) [inline scripts/applications/sequence-picker.mjs]:
```suggestion
      const spr = PIXI.Sprite.from(img);
```

File: scripts/applications/sequence-picker.mjs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      Object.assign(spr, point);
+      spr.anchor.set(0.5);
+      spr.height = spr.width = canvas.grid.size;
+      const sprite = canvas.effects.addChildAt(spr, point);
```
Comment (Freeze023) [inline scripts/applications/sequence-picker.mjs:126]:
```suggestion
      const sprite = canvas.effects.addChildAt(spr, point);
      const mask = new PIXI.Graphics()
        .beginFill("WHITE",1)
        .drawCircle(0, 0, 0.5 * sprite.texture.baseTexture.width)
        .endFill();
      sprite.addChild(mask);
      sprite.mask = mask;
```

File: styles/sequence-picker.css [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/** TARGET SEQUENCE PICKER */
+.sequence-picker {
```
Comment (Freeze023) [inline styles/sequence-picker.css]:
```suggestion
.polo-utils-sequence-picker {
```

File: styles/sequence-picker.css [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  height: auto !important;
+  width: 200px;
+}
+.sequence-picker .link {
```
Comment (Freeze023) [inline styles/sequence-picker.css]:
```suggestion
.polo-utils-sequence-picker .link {
```

File: styles/sequence-picker.css [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  padding: 0.5em 0;
+  text-align: center;
+}
+.sequence-picker .link img {
```
Comment (Freeze023) [inline styles/sequence-picker.css]:
```suggestion
.polo-utils-sequence-picker .link img {
```

File: styles/sequence-picker.css [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  border-radius: 2em;
+  width: 50px;
+  height: 50px;
+  border: none;
+}
+.sequence-picker .link img.inactive {
```
Comment (Freeze023) [inline styles/sequence-picker.css]:
```suggestion
.polo-utils-sequence-picker .link img.inactive {
```

File: styles/sequence-picker.css [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  filter: grayscale(1);
+}
+.sequence-picker .label {
```
Comment (Freeze023) [inline styles/sequence-picker.css]:
```suggestion
.polo-utils-sequence-picker .label {
```

**Review by Freeze023 — COMMENTED** [review]
done.

**Comment by Freeze023** [comment]
clicking the flag in the dialog that was set to unset it, would be a nice feature
or clicking the flag on the canvas

File: lang/en.json [hunk 9]
```diff
@@ -1 +1,4 @@
-{}
\ No newline at end of file
+{
+  "POLO.SequencePicker": "Sequence Picker",
+  "POLO.AllGucci": "All Gucci"
```
Comment (Freeze023) [inline lang/en.json]: We arent really keeping All Gucci are we? Please tell me no, and if it is yes, please lie to me, and still say no.

Comment (krbz999) [reply lang/en.json]: oh right, oops

**Pull request merged**